tests/commands/test_store_client.py

```diff
@@ -135,6 +136,26 @@ class _Client(Client, _StoreClientMock):
     auth_patch.stop()
 
 
+@pytest.fixture
+def anonymized_client_class():
+    """Return a anonymized client instance."""
+    with patch("charmcraft.commands.store.store.AnonymizedClient"):
```

Comment:
this feels like a lot of mocking, it's hard to tell what's going on. Can we mock just `HTTPClient.request()`? It's the only method that's used right?
also you can use the `mocker` fixture to simplify these mocks quite a bit

Reply:
Yes, that can be done